pkg/filesystem/subvolume.go

```diff
@@ -52,6 +60,48 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	cephclusters, _ := clientsets.Rook.CephV1().CephClusters(clusterNamespace).List(ctx, v1.ListOptions{})
+	for _, cephcluster := range cephclusters.Items {
+		enable = cephcluster.Spec.External.Enable
+	}
+	return enable
+}
+
+// getExternalClusterDetails gets the required mon-ip, id and key to connect to the
+// ceph cluster.
+func getExternalClusterDetails(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) (string, string, string) {
+	var admin_id, admin_key, m string
+
+	scList, err := clientsets.Kube.CoreV1().Secrets(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching secrets: %v\n", err))
+	}
+	for _, sc := range scList.Items {
+		if sc.ObjectMeta.Name == "rook-csi-cephfs-node" {
+			data := sc.Data
+			admin_id = string(data["adminID"])
+			admin_key = string(data["adminKey"])
+		}
+	}
+
+	cm, err := clientsets.Kube.CoreV1().ConfigMaps(clusterNamespace).Get(ctx, "rook-ceph-mon-endpoints", v1.GetOptions{})
```

Comment:
This will be in operator namespace or cluster Namespace?

Reply:
It should be cluster Namespace only, @subhamkrai correct me if I am wrong!!

Reply:
This will be clusterNamespace only, only rook-operator related resources require operatorNamespace and this cm is related to cephCLuster.